pkg/kubelet/pleg/generic.go

```diff
@@ -452,6 +494,24 @@ func updateRunningPodAndContainerMetrics(pods []*kubecontainer.Pod) {
 	metrics.RunningPodCount.Set(float64(runningSandboxNum))
 }
 
+// Too many pods in node scene, relist take long time to update pod cache
+// which result in node not ready.
+// readParallelUpdateCacheNum get parallel num to update pod cache.
+// You can set env variable PARALLEL_UPDATE_CACHE_NUM to control it.
+func readParallelUpdateCacheNum() int {
+	num := 20
+	if s := os.Getenv("PARALLEL_UPDATE_CACHE_NUM"); len(s) > 0 {
```

Comment:
Definitely agree that this should be settable. However, I don't have a deep enough knowledge of container runtimes and the impact that this will potentially have on them. IMO, it might make the most sense to have this default to 1 or some other small amount because if it defaults to 1, it will have zero impact and those who are seeing issues with how long it takes to relist will be able to increase that.

As for the name, should be we more explicit what this is for? `PLEG_PARALLEL_UPDATE_CACHE_NUM` or something.

Reply:
agree. see updated.